XenAdmin/Wizards/PatchingWizard/PlanActions/ApplyXenServerPatchPlanAction.cs

```diff
@@ -53,15 +54,23 @@ public ApplyXenServerPatchPlanAction(Host host, XenServerPatch xenServerPatch, L
         protected override void RunWithSession(ref Session session)
         {
 
-            var mapping = mappings.Find(m => m.XenServerPatch.Uuid == xenServerPatch.Uuid && m.MasterHost == Helpers.GetMaster(host.Connection));
+            var master = Helpers.GetMaster(host.Connection);
+            var mapping = mappings.Find(m => m.XenServerPatch.Uuid == xenServerPatch.Uuid && m.MasterHost.uuid == master.uuid);

```

Comment:
The XenServerPatch equality checks further down are inconcistent with this one. Shouldn't you check the patch uuid further down too?
Also, why not use the XenServerPatch.Equals method that does this. You could actaully expand that class with proper operator overloads and Equals overrides so we avoid this error in future.